pure3.md

```diff
@@ -0,0 +1,225 @@
+attempt to secure specific aspects of the software supply chain. For example,
+though in-toto can be used to enforce policies pertaining to the flow of
+artifacts through the supply chain, and can provide proof of the people or bots
+authorized to make changes at each step, it does not intrinsically provide
+mechanisms to store the metadata. Similarly, in-toto does not provide the
+information found in a Software Bill of Materials (SBOM) in a standardized
+manner. With the importance of SBOMs growing in the past few years due to both
+the recognition that knowing the components in a software artifact is vital to
+supply chain security and because they are at the heart of a number of new
+regulations and standards, in-toto is working to develop capabilities that can
+express SPDX and CycloneDX SBOMs as attestations. This means the distinct
+security properties of both in-toto and SBOMs can be leveraged. in-toto is also
+working on capabilities that can be used to either generate complete SBOMs or to
+validate that a provided SBOM is complete.
+
+## Specification
+
+Note: The reader is expected to be familiar with the in-toto specification.
+
+We propose in-toto as an option to achieve software supply chain security
+alongside the Uptane deployment. In such a model, in-toto can be backed by
+the Uptane root of trust, and use Uptane's delegation structure to distribute
+all of in-toto's metadata. Essentially, in-toto metadata is used to attest to
+the steps performed in the software supply chain itself, while Uptane is used
+to securely distribute that metadata alongside the ECU images. This combination
+of in-toto for software supply chain guarantees and Uptane for secure
+distribution of ECU images and in-toto metadata is called Scudo.
+
+### Modifications to the Repositories and Vehicle Interactions
+
+In this document, we propose updating one of Uptane's two repositories to store
+in-toto metadata alongside the images. The Image repository is recommended as
+its contents are typically signed for using offline keys, while the Director
+repository usually uses online keys.
+
+When an image is submitted to the Image repository, it must be accompanied by
+its in-toto metadata. The in-toto layout and link metadata for every image are
+noted in the opaque "custom" field of the image's entry in Targets metadata.
+This way, there is no ambiguity about the in-toto metadata to be used when an
+image is fetched and verified. Further, as it may not be practical for the top
+level Targets role to issue new metadata for every update, Uptane's delegations
+model can be used to make this entire process more scalable. Instead, delegated
+roles can be used to sign for the image itself and its in-toto link metadata.
+However, in-toto layouts and the key artifacts used to verify them should be
+signed for by the top level Targets role.
+
+Once the Director repository decides which images are required and signs Targets
```

Comment:
Please clarify whether or not the Director should also sign any in-toto metadata, particularly for PV secondaries.

Reply:
In Scudo, we haven't modified the functions of the Director repository at all. Should we explicitly state that?